### PR TITLE
Fixes races accessing node and updating it during fingerprinting

### PR DESCRIPTION
**Pull request by dadgar**
Fixes https://github.com/hashicorp/nomad/issues/4154

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -1461,6 +1461,7 @@ func (n *Node) Copy() *Node {
 	nn.Meta = helper.CopyMapStringString(nn.Meta)
 	nn.Events = copyNodeEvents(n.Events)
 	nn.DrainStrategy = nn.DrainStrategy.Copy()
+	nn.Drivers = copyNodeDrivers(n.Drivers)
```
Comment (chelseakomlo) [inline nomad/structs/structs.go]: If there is a test for Copy for a node, add this to the test assertion.

File: client/client.go [hunk 2]
```diff
@@ -1752,9 +1755,14 @@ OUTER:
 	}
 }
 
-// updateNode triggers a client to update its node copy if it isn't doing
+// updateNode updates the Node copy and triggers the client to send the updated
+// Node to the server. This should be done while holding the configLock lock.
 // so already
```
Comment (chelseakomlo) [inline client/client.go]: I don't think this last comment was meant to be included

File: client/client.go [hunk 3]
```diff
@@ -1752,9 +1755,14 @@ OUTER:
 	}
 }
 
-// updateNode triggers a client to update its node copy if it isn't doing
+// updateNode updates the Node copy and triggers the client to send the updated
+// Node to the server. This should be done while holding the configLock lock.
```
Comment (chelseakomlo) [inline client/client.go]: Maybe specify that the caller needs to hold the configLock

Comment (dadgar) [reply client/client.go]: Already states that.

**Comment by dadgar** [comment]
Closing in favor of https://github.com/hashicorp/nomad/pull/4166

**Pull request closed without merging**